From 340c89d2380ffeeb64e169ed53e7a86835f7986b Mon Sep 17 00:00:00 2001
From: Amith A <quic_amitajit@quicinc.com>
Date: Mon, 19 Aug 2024 12:15:48 +0530
Subject: [PATCH] hostapd: Set last octet of country string field to 0x04

In beacon frames, the last octet of the country string field in the
country IE is currently advertising a value of 0x20. However,
according to the WFA Agile Multiband Tech Spec, it should be set to
a value of 0x04. This adjustment is currently made only if the current
operating opclass is a 6 GHz opclass.

Hence set the last octet of the country string to 0x4, irrespective of the
current operating opclass.

Signed-off-by: Amith A <quic_amitajit@quicinc.com>
---
 src/ap/beacon.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/ap/beacon.c b/src/ap/beacon.c
index fd636fe..7eb2366 100644
--- a/src/ap/beacon.c
+++ b/src/ap/beacon.c
@@ -281,11 +281,10 @@ static u8 * hostapd_eid_country(struct hostapd_data *hapd, u8 *eid,
 	os_memcpy(pos, hapd->iconf->country, 3); /* e.g., 'US ' */
 	pos += 3;
 
+	/* Force the third octet of the country string to indicate
+	 * Global Operating Class (Table E-4) */
+	eid[4] = 0x04;
 	if (is_6ghz_op_class(hapd->iconf->op_class)) {
-		/* Force the third octet of the country string to indicate
-		 * Global Operating Class (Table E-4) */
-		eid[4] = 0x04;
-
 		/* Operating Triplet field */
 		/* Operating Extension Identifier (>= 201 to indicate this is
 		 * not a Subband Triplet field) */
-- 
2.34.1

